File: 7kyu/(7 kyu) Form The Minimum/(7 kyu) Form The Minimum.py
# #1
# def min_value(digits):
#     s = set(digits)
#     l = list(s)
#     l.sort()
#     r = []
#     for x in l:
#         r.append(str(x))
#     res = ''.join(r)
#     return int(res)
#2
def min_value(digits):
    l = sorted(list(set(digits)))
    return int(''.join(str(x) for x in l))
#3
def min_value(digits):
     return int("".join(map(str,sorted(set(digits)))))

File: 7kyu/(7 kyu) Form The Minimum/(7 kyu) Form The Minimum.js
// #1
// function minValue(values) {
//   const arr = [...values];
//   arr.sort();
//   const dedupe = arr.filter((x, i, a) => i === a.indexOf(x));
//   const res = dedupe.join('');
//   return parseInt(res);
// }
// #2
// function minValue(values) {
//   const arr = [...values];
//   arr.sort();
//   const dedupe = [...new Set(arr)];
//   const res = dedupe.join('');
//   return parseInt(res);
// }
// #3
// function minValue(values) {
//   const res = [...new Set([...values])].sort().join``;
//   return parseInt(res);
// }
// #4
const minValue = (values) => +[...new Set([...values])].sort().join``;
